File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/SwitchShouldNotBeNested.cs
using System;

namespace Tests.Diagnostics
{
    public class Program
    {
        public Program(char choice, int value)
        {
            switch (choice)
            {
                case 'Y':
                    Console.WriteLine("Yes");
                    break;
                case 'M':
                    Console.WriteLine("Maybe");
                    break;
                case 'N':
                    Console.WriteLine("No");
                    break;
                default:
                    switch (value) // Noncompliant
//                  ^^^^^^
                    {
                        case 0:
                            break;
                        default:
                            break;
                    }
                    Console.WriteLine("Invalid response");
                    break;
            }


            int i = 1;
            switch (i)
            {
                case 1:
                case 2:
                    Console.WriteLine("One or Two");
                    break;
                default:
                    Console.WriteLine("Other");
                    break;
            }

            string first = "foo", second = "bar";

            var result = first switch
            {
                "a" => second switch // Compliant
                {
                    "x" => 9,
                    _ => 1
                },
                "b" => 2,
                _ => 3
            };
        }
    }
}

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SwitchShouldNotBeNested.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace Tests.TestCases
Public Class Program
    Public Sub New()
        Dim choice As String = "Foo"
        Dim value As Integer = 1

        Select Case choice
            Case "Y"c
                Console.WriteLine("Yes")
            Case "M"c
                Console.WriteLine("Maybe")
            Case "N"c
                Console.WriteLine("No")
            Case Else

                    Select Case value ' Noncompliant {{Refactor the code to eliminate this nested 'Select Case'.}}
'                   ^^^^^^^^^^^^^^^^^
                        Case 0
                        Case Else
                    End Select

                    Console.WriteLine("Invalid response")
        End Select

        Dim i As Integer = 1

        Select Case i
            Case 1, 2
                Console.WriteLine("One or Two")
            Case Else
                Console.WriteLine("Other")
        End Select
    End Sub
End Class

End Namespace
